### ISSUE TITLE
Feedback welcome on Devt branch

### ISSUE THREAD

**Issue by luc-github** [luc-github/ESP3D#168]

**Comment by benji720** [comment]
Hi - I've done a few changes to incorporate % complete and planning IFTTT integration.  I've got no idea how to merge into GitHub (I don't normally use it).  Do you have that in your branch?  Do you want it (% complete is trivial - just issue a M27 every so often and include the output in the STATUS JSON)?

I was also going to update the web interface - I think it can look much nicer.

**Comment by luc-github** [comment]
Yes ESP3D-WEBUI will become the new UI but need Devt branch to work - it will be merged in master end of this month

**Comment by Sparkxxx** [comment]
Hello Mr. Lebosse, 

While testing and playing with the development version I enabled AUTHENTICATION_FEATURE in config.h and since then the web interface is just displaying Connection failed! is your FW correct? 
Digging deper in the code I stumbled on the browser console log that sais 
Error 403 : Not allowed, log in first! 
so my assumption is that something wrong is happening since there's no way I can login at this step since I am just loading for the first time.
Disabling the authentification renders the page normally.

**Comment by luc-github** [comment]
Hi Thank you for this feedback - yes, you are correct, current UI does not have any UI for authentication but FW itself is in theory ready -  I will implement soon a login page and a change password page for user and admin

I am currently working on first setup steps pages then will work on authentication, I wanted to get full process working without authentication before enabling it

**Comment by apphead** [comment]
Using devt version, when these feature are uncommented in config.h, they throw errors:
- AUTHENTICATION_FEATURE: as reported above by Sparkxxx
- SDCARD_FEATURE: compilation fails due to
....arduino_build_743302\sketch\config.h:169:19: fatal error: SdFat.h: No such file or directory
 #include "SdFat.h"
File is present in _[ESP3D-devt]\libraries\SdFat\SdFat\src\_

**Comment by luc-github** [comment]
Thanks for the feedback 

SDCARD_FEATURE is not available yet, that is why it is currently commented
to get SdFat.h you need to install the library manually https://www.arduino.cc/en/Guide/Libraries#toc5
But I did not documented it yet as SDCARD_FEATURE  is not available - but will be available soon (code is ready but not public)

**Comment by luc-github** [comment]
Authentication is now implemented in FW and UI : https://github.com/luc-github/ESP3D/commit/798f6ea149c8e5877891531a94d352765646d400

**Comment by eduard-sukharev** [comment]
Seems like Sd card code is here now: i've tried building from master and overall firmware looks and works ok, but uploading files to sd card works wrong. I created a simple gcode file with only G28 command in it and tried to upload it, but the printer behaves as if it was given direct gcode commands, three times in a row! It homes all axes, then homes them again and then again.
Is it a bug and should i report it as a separate issue?

It's Marlin fw, fresh esp8266 and esp3d.

**Comment by luc-github** [comment]
you mean serial sd upload ?

**Comment by eduard-sukharev** [comment]
@luc-github i mean the ability to transfer files to printer's SD card over wifi.
Here's the settings page of my FW:
![fw_settings](https://user-images.githubusercontent.com/3235378/30923527-bf3bc8fa-a3b4-11e7-95cb-e9001a85bdbd.png)

My setup is following: I have MKS Gen v1.4 board (RAMPS 1.4 compatible) with attached ULTIPANEL (which is also an MKS MINI12864 LCD screen with built-in SD card reader). The esp is connected to printer via Tx/Rx, with no additional commutation between Esp module and SD card reader.

**Comment by luc-github** [comment]
Ok So yes it is Serial file transfert
Upload send file to ESP which foward to RAMPS  by serial which write data to the  MKS MINI12864 LCD screen

Does content of SD card is well displayed in UI before upload ?

**Comment by eduard-sukharev** [comment]
Well, kind of. Filenames are in 8.3 DOS-like format with tildas, but i could tell them apart.

**Comment by luc-github** [comment]
yes 8.3 format is the marlin way to display files, you will have same output when doing M20 command in command panel

**Comment by luc-github** [comment]
what file name is used for upload ?

**Comment by eduard-sukharev** [comment]
The filename is 'homing.gcode', the contents is single line "G28" and a newline at the end.

**Comment by eduard-sukharev** [comment]
Is there any debug info i could provide? Or probably some simple firmware sketch to replicate the file uploading to marlin? Is there something in marlin that should be enabled for serial sd card upload?

**Comment by eduard-sukharev** [comment]
I've tested a bit more with different file contents. It does not depend on the new line characters: neither CRLF, nor CR, nor LF work. Comments in gcode lines are also play no role.
The "triple" execution is only for G28. The fllowing file contents:
```
G90 ; use absolute positioning for the XYZ axes
G1 X100

G28 ;

```
makes my printer move the caret once, but then it homes and rehomes and rehomes and rehomes.

After everything, LCD screen shows "SD upload failed".

**Comment by luc-github** [comment]
I suspect the filename - I cannot reproduce now asI am travelling ut this this my guess

the filename is not 8.3 so the command to start writing to SD including filename is not taken in account 
and the content of file is no more see as content but as commands itself.

please try renaming the file in 8.3 like homing.gco and try upload

**Comment by eduard-sukharev** [comment]
The naming of files indeed matters. I've tried renaming a file into 'homing.gco' and 'homing.g' and 'home.g' as you proposed and all uploaded well!

But now i've just found out that i can't start printing of any of the files on sd card!
I have both non-8.3 files and 8.3 files on sd card and when i click the priner button near them the printer does nothing, lcd shows nothing and "Information" logs show following:
```
:Now fresh file: /HOME.G 21
:Now fresh file: /HOMING.GCO 21
:Now fresh file: /HOME-8.GCO 21
:Now fresh file: /HOMING.GCO 21
```
Each line resembles one button click, one print attempt. Any ideas what's wrong?

**Comment by eduard-sukharev** [comment]
Is there any reason why filesize is not split away from filename for Marlin firnware? For repetier and others you seem to split the contents of json by space character, while maelin filenames are left intact with numbers after space. I suspect that might be the reason for prints not starting.

**Comment by luc-github** [comment]
humm this has been fixed in master if I remember well, I may missed to update devt, I will check tonight and add a check for file name lenght

**Comment by luc-github** [comment]
yes issue is in webui and seems I wiped the fix...
21cb4bb701afa95e522c5a91d30b9233d0eefc40

**Comment by eduard-sukharev** [comment]
ah, so to fix this i have to upload web-ui files with current esp3d fw? ok then.
How often do you update UI files in this repo from WEB-UI repo?
And one more question - why file sizes are not split from filenames for marlin?

**Comment by luc-github** [comment]
The fix is to upload the new espwebui when I put fix back
I update webui when there are issues but I am currentling traveling so ...
Before Marlin do not have file size, then they introduced it like others FW but I do not use Marlin so I do not follow all update as soon they are out

**Comment by eduard-sukharev** [comment]
hm, so you didn't wipe the fix. it's there. The problem seems to be that i haven't used espwebui files at all and just uploaded esp3d files as is. Will try espwebui this evening and will give the feedback.
Btw, regarding espwebui - is installation all about uploading `index.html.gz` and that's it? Should no other files be uploaded? No js files? no css or other files?

**Comment by luc-github** [comment]
yes that is all index.html.gz is the only needed file for devt branch

**Comment by luc-github** [comment]
please be sure you use devt branch : https://github.com/luc-github/ESP3D/tree/devt/esp3d

**Comment by mzeve** [comment]
Hey, I cant compile the devt branch as it fails with the error: 'class HardwareSerial' has no member named 'setRxBufferSize' . I dont understand why thats happening

**Comment by luc-github** [comment]
@mzeve devt branch need master git from ESP core not released version, setRxBufferSize is not in current released version
